staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -320,19 +320,15 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 	newDeltas := append(f.items[id], Delta{actionType, obj})
 	newDeltas = dedupDeltas(newDeltas)
 
-	_, exists := f.items[id]
-	if len(newDeltas) > 0 {
-		if !exists {
-			f.queue = append(f.queue, id)
-		}
-		f.items[id] = newDeltas
-		f.cond.Broadcast()
-	} else if exists {
-		// We need to remove this from our map (extra items
-		// in the queue are ignored if they are not in the
-		// map).
-		delete(f.items, id)
+	// newDeltas will always have at least one delta, if there is a case where
+	// it may have no elements, delete f.items[id] (f.queue will ignore elements
```

Comment:
This comment is confusingly worded, as it's talking about a case that is no longer handled here without making that clear.

Reply:
Will rephrase if this change is accepted.